### PR TITLE
Optimize Referenced::Signature function

### PR DESCRIPTION
**Pull request by qiwzhang**
**What this PR does / why we need it**:

Use extra loop to optimize signature calculation.

**Release note**:

```release-note
None
```

### CODE REVIEW

File: src/istio/mixerclient/referenced.cc [hunk 1]
```diff
@@ -117,8 +117,20 @@ bool Referenced::Fill(const Attributes &attributes,
 bool Referenced::Signature(const Attributes &attributes,
                            const std::string &extra_key,
                            std::string *signature) const {
-  const auto &attributes_map = attributes.attributes();
+  if (!CheckAbsenceKeys(attributes)) {
```
Comment (lizan) [inline src/istio/mixerclient/referenced.cc]: Combine two ifs

File: src/istio/mixerclient/referenced.cc [hunk 2]
```diff
@@ -142,27 +154,64 @@ bool Referenced::Signature(const Attributes &attributes,
...
     const auto it = attributes_map.find(key.name);
-    // if an "exact" attribute not present, return false for mismatch.
+    // If an "exact" attribute not present, return false for mismatch.
     if (it == attributes_map.end()) {
       return false;
     }
 
+    const Attributes_AttributeValue &value = it->second;
+    if (value.value_case() == Attributes_AttributeValue::kStringMapValue) {
+      std::string map_key = key.map_key;
```
Comment (lizan) [inline src/istio/mixerclient/referenced.cc]: const &

File: src/istio/mixerclient/referenced.h [hunk 3]
```diff
@@ -49,6 +49,17 @@ class Referenced {
   std::string DebugString() const;
 
  private:
+  // Return true if all absent keys are not in the attributes.
+  bool CheckAbsenceKeys(const ::istio::mixer::v1::Attributes &attributes) const;
```
Comment (JimmyCYJ) [inline src/istio/mixerclient/referenced.h]: s/CheckAbsenceKeys/CheckAbsentKeys ?

File: src/istio/mixerclient/referenced.cc (hunk 2 above)
Comment (qiwzhang) [reply src/istio/mixerclient/referenced.cc]: Could not,  map_key is changed later.

File: src/istio/mixerclient/referenced.h (hunk 3 above)
Comment (qiwzhang) [reply src/istio/mixerclient/referenced.h]: Done

**Pull request merged**